src/test/metric_config_test.cpp

```diff
@@ -0,0 +1,167 @@
+
+using testing::_;
+using testing::Return;
+
+static const char* modelDefaultConfig = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"latest": {"num_versions":1}},
+                "nireq": 100,
+                "shape": {"b": "(1,10) "}
+            }
+        }
+    ]
+})";
+
+static const char* modelMetricsChangedConfig = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"latest": {"num_versions":1}},
+                "nireq": 100,
+                "shape": {"b": "(1,10) "}
+            }
+        }
+    ],
+    "monitoring":
+        {
+            "metrics":
+            {
+                "enable" : true,
+                "metrics_list": ["requestSuccessGrpcPredict", "requestFailRestModelStatus"],
+                "endpoint_path": "/newmetrics" 
+            }
+        }
+})";
+
+static const char* modelMetricsAllEnabledConfig = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"latest": {"num_versions":1}},
+                "nireq": 100,
+                "shape": {"b": "(1,10) "}
+            }
+        }
+    ],
+    "monitoring":
+        {
+            "metrics":
+            {
+                "enable" : true
+            }
+        }
+})";
+
+class MetricsConfigTest : public TestWithTempDir {
+public:
+    std::string configFilePath;
+    std::string ovmsConfig;
+    std::string modelPath;
+    std::string dummyModelName;
+
+    void SetUpConfig(const std::string& configContent) {
+        ovmsConfig = configContent;
+        dummyModelName = "dummy";
+        const std::string modelPathToReplace{"/ovms/src/test/dummy"};
+        ovmsConfig.replace(ovmsConfig.find(modelPathToReplace), modelPathToReplace.size(), modelPath);
+        configFilePath = directoryPath + "/ovms_config.json";
+    }
+    void SetUp() override {
+        TestWithTempDir::SetUp();
+        // Prepare manager
+        modelPath = directoryPath + "/dummy/";
+        SetUpConfig(modelDefaultConfig);
+        std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    }
+};
+
+TEST_F(MetricsConfigTest, DefaultValues) {
+    ConstructorEnabledModelManager manager;
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+
+    auto modelInstance = manager.findModelInstance(dummyModelName);
+    auto modelConfig = modelInstance->getModelConfig();
```

Comment:
Do not copy modelinstance/modelconfig - use reference?

Reply:
Done.